x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -516,7 +516,7 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
 
         // This is an exceptionally weird state
         // Our view of the memory is stale or we have tasks where the required job memory is 0, which should be impossible
-        if (largestJob == 0L && ((dataframeAnalyticsTasks.isEmpty() || anomalyDetectionTasks.isEmpty()) == false)) {
+        if (largestJob == 0L && (dataframeAnalyticsTasks.size() + anomalyDetectionTasks.size() > 0)) {
```

Comment:
It is easier to grok what this is checking with this math on the size I think.

Other wise we get longer nested `&&` checks against empty and more `== false` ,etc.